### ISSUE TITLE
Handle reordering of mergeable commits

### ISSUE THREAD

**Issue by GregoRCodi** [ejoffe/spr#29]
Hey! I'm using spr and I have encountered a problem with reordering commits that were mergeable. I had a stack of 4 commits, the 2nd and the 4th one were approved, but the 1st and 3rd were not.

```sh
$ git spr -s
[✔✔✔✗] commit D
[·✗✔✗] commit C
[✔✔✔✗] commit B
[·✗✔✗] commit A
```

I ran `git rebase -i` and reordered them, so they were in the following order:
1. B
2. D
3. A
4. C

and then I wanted to update the PRs to then merge commits B and D. Howeever, after running `git spr -u`, I got the following error:

```sh
$ git spr -u
panic: Cannot change the base branch of a closed pull request.

goroutine 1 [running]:
github.com/ejoffe/spr/spr.check(...)
        /Users/eitan/Code/spr/spr/spr.go:896
github.com/ejoffe/spr/spr.updateGithubPullRequest(0x7d7a30, 0xc00001a0c8, 0xc00000e0f8, 0xc0004e40a0, 0xc0003a1e40, 0xc00030cd1c, 0x8, 0xc00030cc70, 0x28, 0xc00030ccea, ...)
        /Users/eitan/Code/spr/spr/spr.go:738 +0x3ee
github.com/ejoffe/spr/spr.(*stackediff).UpdatePullRequests(0xc000026680, 0x7d7a30, 0xc00001a0c8)
        /Users/eitan/Code/spr/spr/spr.go:105 +0x2e5
main.main()
        /Users/eitan/Code/spr/cmd/spr/main.go:86 +0x747
```

The result was that PRs for commits B and D were merged, but the commits were not reordered, so I ended up with 2 open PRs, one of which had 3 commits, and another one had 1 commit:
1. PR with B, D, A
2. PR with C

Ideally, `git spr -u` would only rebase those PRs so the order was correct, and not merge anything.

**Comment by ejoffe** [comment]
Thank you for bringing this issue up, this is indeed a bug. 
I have also encountered a similar interesting case when reordering commits.
I'm surprised the PRs merged, because there is not a single call in the update flow to the github merge api. My guess is that somehow changing the PR base and head ref branch causes it to be merged in some cases. Once the PR merges into each other, the situation is hopeless and we are bound to hit a panic like the one you have above. 
I will investigate this issue further.

**Comment by GregoRCodi** [comment]
I see... Thanks for taking the time to investigate :+1: I appreciate it

**Comment by ejoffe** [comment]
what version you are using? 'git spr -v'

**Comment by GregoRCodi** [comment]
```sh
14:20 $ git spr -v
spr version : 0.4.3 : 2021-06-08T18:26:00Z : 8a7a9448
```

**Comment by ejoffe** [comment]
After further investigation, force pushing the commits to github when they are reordered is causing the pull requests to automagically merge into each other. The solution I have found for this case, is in the case of commit reorder, retarget all the pull requests to the master branch, followed by pushing the newly ordered commits, followed by updating the pull requests target branch to reorder the pull requests. 
With this fix, I have been able to do multiple complicated commit reorderings in a test setup with no issues.
